### ISSUE TITLE
Search Index: Filtering out characters

### ISSUE THREAD

**Issue by Pete-Robelou** [ceph/ceph.io#118]
To prevent the search index from falling over, which can happen if there is an incompatible character in the article content, we should implement a stricter approach in our `cleanIndex` filter and remove more characters.

It's probably best to go wide with this approach as we won't know for sure what content will be in an article.

**Comment by Pete-Robelou** [comment]
@adamduncan I've renamed our `cleanIndex` filter to `cleanSearchRaw` as it's tailored to that exact purpose.

I've tweaked some of the regex declarations and refactored so it's a little easier to read. We also now remove unicode characters as these were causing issues.

We also now have a `cleanSearchOutput` filter. I created this to clean the search data we use for the search result cards as they would fall foul of the same issues as the raw data. Things like control characters, line breaks,  unicode etc... still need to be filtered but it's different to the raw search data in that it doesn't need to make the text lower case, remove duplicate or short, less meaninful words and certain bits of punctuation. These changes make it different enough from `cleanSearchRaw` so I created a separate filter.
